src/main/java/duke/program/TaskManager.java

```diff
@@ -0,0 +1,404 @@
+
+    private static final String SEPARATOR_SPACE = " ";
+    private static final String SEPARATOR_DOT = ".";
+    private static final String SEPARATOR_DOT_WITH_ESCAPE = "\\.";
+    private static final String SEPARATOR_BY = "/by";
+    private static final String SEPARATOR_AT = "/at";
+
+    private static final String TASK_TYPE_ICON_TODO = "T";
+    private static final String TASK_TYPE_ICON_DEADLINE = "D";
+    private static final String TASK_TYPE_ICON_EVENT = "E";
+    private static final String ICON_DONE = "X";
+
+    private static final int TODO_DESCRIPTION_START_INDEX = 5;
+    private static final int DEADLINE_DESCRIPTION_START_INDEX = 9;
+    private static final int EVENT_DESCRIPTION_START_INDEX = 6;
+    private static final int LINE_LENGTH = 40;
+
+    private ArrayList<Task> tasks;
+    private int taskCount;
+
+    public TaskManager() {
+        this.tasks = new ArrayList<>();
+        this.taskCount = 0;
+    }
+
+    public static void printLine() {
+        for (int i = 0; i < LINE_LENGTH; i++) {
+            System.out.print("_");
+        }
+        System.out.println("");
+    }
+
+    public static void printIndexOutOfBoundsMessage() {
+        System.out.println("Sorry bud, you can't check off/delete what is not yet there :/");
+    }
+
+    public static void printInvalidDoneOrDeleteMessage() {
+        System.out.println("Sorry bud, that's not a valid task number to check off/delete!");
+    }
+
+    public static void printAlreadyDoneMessage() {
+        System.out.println("Slow down there bud! You've already completed this task!");
+    }
+
+    public static void printEmptyIndexAfterDoneMessage() {
+        System.out.println("Hey bud, the format for marking off a task is :done [index]");
+    }
+
+    public static void printEmptyIndexAfterDeleteMessage() {
+        System.out.println("Hey bud, the format for marking off a task is :delete [index]");
+    }
+
+    public static void printEmptyDescriptionMessage(String command) {
+        System.out.println("Sorry bud! The description after " + command + " can't be blank!");
+    }
+
+    public static void printInvalidCommandMessage() {
+        System.out.println("Sorry bud, but that command is gibberish to me. I can only read 7 words!");
+        System.out.println("The seven words are:");
+        System.out.printf("list%ndone%ndelete%ntodo%ndeadline%nevent%nbye%n");
+    }
+
+    public static void printInvalidFormatMessage(String command) {
+        String timeIndicatorCommand = (command.equals(COMMAND_DEADLINE)) ? "by" : "at";
+        System.out.println("Sorry bud, but your formatting is gibberish to me. Here is how it should be formatted: ");
+        System.out.println(command + " {description} /" + timeIndicatorCommand + " {timing}");
+    }
+
+    public static void printGenericErrorMessage() {
+        System.out.println("Oops! Something went wrong :(");
+    }
+
+    public void printMarkAsDoneMessage(Task task) {
+
```

Comment:
This space after method name is inconsistent with other methods. You could remove it.